pipeline_dp/accumulator.py

```diff
@@ -187,15 +188,17 @@ def compute_metrics(self) -> float:
 
 @dataclass
 class VectorSummationParams:
-    noise_kind: aggregate_params.NoiseKind
+    dp_params: dp_computations.MeanVarParams
+    clip_norm_kind: str
     max_norm: float
 
 _FloatVector = Union[Tuple[float], np.ndarray]
 
 class VectorSummationAccumulator(Accumulator):
     _vec_sum: np.ndarray
+    _params: Optional[VectorSummationParams]
 
-    def __init__(self, params: VectorSummationParams,
+    def __init__(self, params: Optional[VectorSummationParams],
```

Comment:
@dvadym 
Does the `Accumulator` API always assume that we are computing over private units? If my understanding is correct, we can have the next query:
```sql
SELECT public_partition1, vec_sum_accum(@params_outer, vals[:, 1:])
FROM (
    SELECT public_partition1, public_partition2, 
            some_nonlinear_function(vec_sum_accum(@params_inner, vals)) as vals
    FROM data
    GROUP BY public_partition1,public_partition2
)
GROUP BY public_partition1
```
In such a case - the inner `vec_sum_accum` will have `@params_inner` which make the accumulation `(eps, delta)`-DP, but the **outer** accumulation doesn't have to use anonymization to satisfy (e,d)-DP according to the Post Processing Rule.  
Such a query is definitely possible and in-scope of data engineering and processing.  
For instance, in a context of recommendation systems: If we have the full unique key of a dataset consist of `user_id, product_id`, and `vals` is some vector of continuous values that have information over some "preference parameters" for this user. The recommendation system will attempt to rank all the `product_id`s per user, using some Machine Learning technique, and somewhere in this model we'll have a reduce-broadcast (e.g. softmax). The inner sum will have to ensure `(eps, delta)`-DP, but this inner sum is then passed forward to the next computation that might also include reduce-broadcast. The second reduce-broadcast now doesn't require anonymization, becaues it's Post Processing on anonymized data.

Reply:
Nested queries is a valid usecase in data processing, though it's outside of our scope - implementing DP computations. It's up to the user to post-process data in any way they like. Maybe in future we decide to provide APIs for post-processing, but it's hard to foresee future :). So it's better to go simple and always do DP. Minimizing functionality of any specific class makes things simpler, more testable.

General considerations about PipelineDP API. It's usually a good idea to minimize functionality exposed to the user since user visible API is hard to change/remove (users might depend on it) and there is a maintenance cost to keep API not-changed (because the implementation usually evolves). And it's always possible to add some functionality in the public API (and implementation) later. For some additional functionality (as post-processing), it's what users might do themselves. And if there are frequent requests from them we can think how to implement that (and it would be easier, because we will know use cases and will have more experience).

Reply:
Agreed, Done.